rpcserver.go

```diff
@@ -5340,6 +5340,78 @@ func (r *rpcServer) InsertProof(ctx context.Context,
 	return r.marshalUniverseProofLeaf(ctx, req.Key, newUniverseState)
 }
 
+// PushProof attempts to query the local universe for a proof specified by a
+// UniverseKey. If found, a connection is made to a remote Universe server to
+// attempt to upload the asset leaf.
+func (r *rpcServer) PushProof(ctx context.Context,
+	req *unirpc.PushProofRequest) (*unirpc.PushProofResponse, error) {
+
+	switch {
+	case req.Server == nil:
+		return nil, fmt.Errorf("remote Universe must be specified")
+
+	case req.Key == nil:
+		return nil, fmt.Errorf("Universe key must be specified")
+	}
+
+	switch {
+	case req.Server.Host == "" && req.Server.Id == 0:
```

Comment:
I think Golang switch statements are ordered and the first that matches is executed. So could combine this with the above.